src/script.rs

```diff
@@ -403,6 +392,72 @@ impl ScriptNum {
     }
 }
 
+impl From<i64> for ScriptNum {
+    fn from(value: i64) -> Self {
+        ScriptNum(value)
+    }
+}
+
+impl From<i32> for ScriptNum {
+    fn from(value: i32) -> Self {
+        ScriptNum(value.into())
+    }
+}
+
+impl From<u8> for ScriptNum {
+    fn from(value: u8) -> Self {
+        ScriptNum(value.into())
+    }
+}
+
+/// TODO: This instance will be obsolete if we convert bool directly to a `Vec<u8>`, which is also
+///       more efficient.
+impl From<bool> for ScriptNum {
+    fn from(value: bool) -> Self {
+        ScriptNum(value.into())
+    }
+}
+
+impl TryFrom<usize> for ScriptNum {
+    type Error = TryFromIntError;
+    fn try_from(value: usize) -> Result<Self, Self::Error> {
+        value.try_into().map(ScriptNum)
+    }
+}
+
+/// This is the primary extractor for numbers from `Script`, all other extractors should call this.
+impl From<ScriptNum> for i32 {
+    fn from(value: ScriptNum) -> Self {
+        if value.0 > i32::MAX.into() {
+            i32::MAX
+        } else if value.0 < i32::MIN.into() {
+            i32::MIN
+        } else {
+            value.0.try_into().unwrap()
```

Comment:
Is the idea here that a `ScriptNum` can represent a value that is out of range internally through various computations, and is only clamped when converting back to an `i32`? In this particular case, I would have made this an `as_i32` method on `ScriptNum` instead of a `From` instance.

Reply:
Exactly. So this was previously `getint()`, but to simplify the various conversions mentioned in https://github.com/ZcashFoundation/zcash_script/pull/174#discussion_r1927796619 (and other locations), I added `TryFrom<ScriptNum> for u8`/`for u16`. I figured the `From<ScriptNum> for i32` was more consistent. However, slightly oddly, we never actually extract an `i32` from a `ScriptNum`, so reverting this instance to `getint` wouldn’t change anything other than those `TryFrom` instances.